app/models/vm_or_template.rb

```diff
@@ -1939,5 +1939,14 @@ def self.arel_coalesce(values)
     Arel::Nodes::NamedFunction.new('COALESCE', values)
   end
 
+  # TODO: add states for other vms
+  def self.calculate_raw_power_state(state)
```

Comment:
this probably needs to be down on `vm` and not up here on `vm_or_template` ... maybe

Templates technically shouldn't have a raw_power_state.